src/site/content/en/blog/module-workers/index.md

```diff
@@ -0,0 +1,162 @@
+  postMessage(sayHello());
+});
+```
+
+##### greet.js:
+
+```js
+// global to the whole worker
+function sayHello() {
+  return 'world';
+}
+```
+
+For this reason, using Web Workers has historically imposed an outsized effect on the architecture of an application. Developers have had to create clever tooling and workarounds to make it possible to use Web Workers without giving up modern development practises. As an example, bundlers like Webpack embed a small module loader implementation into generated code that uses `importScripts` for code loading, but wraps modules in functions to avoid variable collisions and simulate dependency imports and exports.
+
+
+## Enter Module Workers
+
+A new mode for Web Workers with the ergonomics and performance benefits of [JavaScript Modules](https://v8.dev/features/modules) is shipping in Chrome 80, called Module Workers. The Worker constructor accepts a new `{type:"module"}` option, which changes script loading and execution to match `<script type="module">`.
+
+```js
+const worker = new Worker('worker.js', {
+  type: 'module'
+});
+```
+
+Since Module Workers are JS Modules, they can use standard JavaScript import and export statements. As with all JS Modules, dependencies are only executed once in a given context (main thread, worker, etc), and all future imports reference the already-executed module instance. The loading and execution of JS Modules is also optimized by browsers - because a Module's dependencies can be loaded prior to the Module being executed, entire module trees can be loaded in parallel. Module loading also caches parsed code, which means modules that are used on the main thread and in a Worker only need to be parsed once.
+
+Moving to JS Modules also enables the use of [dynamic import](https://v8.dev/features/dynamic-import) for lazy-loading code without blocking execution of the worker. Dynamic Import is much more explicit than using importScripts to load dependencies, since the imported module's exports are returned rather than relying on global variables.
+
+##### worker.js:
+
+```js
+import { sayHello } from './greet.js';
+addEventListener('message', e => {
+  postMessage(sayHello());
+});
+```
+
+##### greet.js:
+
+```js
+import greetings from './data.js';
+export function sayHello() {
+  return greetings.hello;
+}
+```
+
+To ensure great performance, the old `importScripts()` API is not available within Module Workers. Switching Workers to use JS Modules means all code is loaded in [strict mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode), and HTML-style comments are not supported. Another notable change is that the value of `this` in the top-level scope of a JS Module is `undefined`, whereas in Classic Workers the value was the worker's global scope. Fortunately, there has always been a `self` global that provides a reference to the global scope - it's available in all types of Workers including Service Workers, as well as in the DOM.
```

Comment:
What's the significance of "HTML-style comments" not being supported? I'm having a hard time imagining how HTML-style comments would have been used in an classic worker—was that really a thing?

Reply:
I believe they were supported, though I can't imagine anyone has ever used them. Maybe for IE conditional comments?

<img src="https://user-images.githubusercontent.com/105127/70931325-2bccf880-2005-11ea-9175-35d9442c2ba2.png" width="260">

Reply:
Weird.

I'd consider just dropping mention of that, or move it to an aside or something? For an edge case, it's needlessly puzzling.